casepro/msgs/models.py

```diff
@@ -132,6 +133,125 @@ def __str__(self):
 
 
 @python_2_unicode_compatible
+class Language(models.Model):
+    """
+    Languages used, defined by the ISO 639-3 language code combined with the location code, e.g. eng_UK
+    """
+    org = models.ForeignKey(Org, verbose_name=_("Organization"), related_name='languages')
+
+    code = models.CharField(max_length=6)  # e.g. eng_UK
+
+    name = models.CharField(max_length=100, null=True, blank=True)  # e.g. English
+
+    location = models.CharField(max_length=100, null=True, blank=True)  # e.g. United Kingdom
+
+    @classmethod
+    def search(cls, org, user, search):
+        """
+        Search for Languages
+        """
+        name = search.get('name')
+        location = search.get('location')
+
+        queryset = Language.objects.all()
+
+        # Name filtering
+        if name:
+            queryset = queryset.filter(name__icontains=name)
+
+        # Location filtering
+        if location:
+            queryset = queryset.filter(location__icontains=location)
+
+        return queryset.order_by('code')
+
+    def as_json(self):
+        return {
+            'id': self.pk,
+            'code': self.code,
+            'name': self.name,
+            'location': self.location
+        }
+
+    def __str__(self):
+        return "%s" % self.code
+
+
+@python_2_unicode_compatible
+class FAQ(models.Model):
+    """
+    Pre-approved questions and answers to be used when replying to a message.
+    """
+    org = models.ForeignKey(Org, verbose_name=_("Organization"), related_name='faqs')
+
+    question = models.CharField(max_length=255)
+
+    answer = models.TextField()
+
+    language = models.ForeignKey(Language, verbose_name=('Language'), related_name='faqs', default=None)
+
+    parent = models.ForeignKey('self', null=True, blank=True, related_name='translations')
```

Comment:
Not sure about this parent/child relationship between translations. Seems like it would be simpler to have a single `FAQ` instance for each question, and then something like an HSTORE to hold all of that FAQ's different translations, no?

Also don't think `blank=True` does anything on `ForeignKey` fields